fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/temp/properties/RelationalExpressionDepthProperty.java

```diff
@@ -90,4 +99,11 @@ public int evaluate(@Nonnull RelationalExpression expression) {
         return expression.acceptPropertyVisitor(this);
     }
 
+    @Nonnull
+    @Override
+    @SpotBugsSuppressWarnings("NP_PARAMETER_MUST_BE_NONNULL_BUT_MARKED_AS_NULLABLE")
+    public Integer evaluateAtQuantifier(@Nonnull final Quantifier quantifier, @Nullable final Integer rangesOverResult) {
+        // since we do visit expression references in this property we can insist on rangesOverResult not being null
```

Comment:
```suggestion
        // since we visit the expression reference under the quantifier, and don't return null ourselves, we can
        // insist that rangesOverResult is never null
```